app/services/visit/integrity_checker.rb

```diff
@@ -0,0 +1,56 @@
+class Visit
+  module IntegrityChecker
+    def ensure_visit_integrity
+      return if required_information?
+      log_any_missing_information
+      flash[:notice] = I18n.t(
+        :ensure_visit_integrity,
+        scope: 'controllers.shared'
+      )
+      redirect_to edit_prisoner_details_path
+    end
+
+    def required_information?
+      return false unless visit
+      return false unless visit.prisoner?
+      return false unless visit.prison_name?
+      true
```

Comment:
Isn't this just `visit && visit.prisoner? && visit.prisoner_name?`? All the `false unless` makes my head hurt, though, so I may be missing some wrinkle.